projects/rocsparse/clients/benchmarks/CMakeLists.txt

```
@@ -1,252 +1,82 @@
-../testings/testing_dense_to_sparse_csc.cpp
-../testings/testing_sparse_to_sparse.cpp
-../testings/testing_extract.cpp
-../testings/testing_spgemm_bsr.cpp
-../testings/testing_spgemm_csr.cpp
-../testings/testing_spgemm_reuse_csr.cpp
-../testings/testing_spgeam_csr.cpp
-../testings/testing_spgeam_reuse_csr.cpp
-../testings/testing_spgeam_csr_2.cpp
-../testings/testing_spgeam_reuse_csr_2.cpp
-../testings/testing_gemvi.cpp
-../testings/testing_sddmm.cpp
-../testings/testing_csrcolor.cpp
-../testings/testing_check_matrix_csr.cpp
-../testings/testing_check_matrix_csc.cpp
-../testings/testing_check_matrix_coo.cpp
-../testings/testing_check_matrix_gebsr.cpp
-../testings/testing_check_matrix_gebsc.cpp
-../testings/testing_check_matrix_ell.cpp
-../testings/testing_check_matrix_hyb.cpp
-../testings/testing_check_spmat.cpp
-../testings/testing_bsrpad_value.cpp
+target_include_directories(rocsparse-bench PRIVATE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
 )
 
-add_executable(rocsparse-bench ${ROCSPARSE_BENCHMARK_SOURCES} ${ROCSPARSE_CLIENTS_COMMON} ${ROCSPARSE_CLIENTS_TESTINGS})
-
-# Target compile options
-target_compile_options(rocsparse-bench PRIVATE -Wno-deprecated -Wno-unused-command-line-argument -Wall)
-
-# Internal common header
-target_include_directories(rocsparse-bench PRIVATE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>)
-if (BUILD_CLIENTS_ONLY)
-  target_include_directories(rocsparse-bench PRIVATE ${rocsparse_INCLUDE_DIRS}/rocsparse)
+# OpenMP RPATH handling for hip-clang
+if(NOT WIN32 AND OPENMP_FOUND)
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+        if(TARGET OpenMP::OpenMP_CXX)
+            set_target_properties(rocsparse-bench PROPERTIES
+                BUILD_RPATH "${HIP_CLANG_ROOT}/lib"
+                INSTALL_RPATH "$ORIGIN/../llvm/lib"
+            )
+        elseif(TARGET OpenMP::omp)
+            if(DEFINED openmp_LIB_DIR)
+                set_target_properties(rocsparse-bench PROPERTIES
+                    BUILD_RPATH "${HIP_CLANG_ROOT}/${openmp_LIB_DIR}"
+                    INSTALL_RPATH "$ORIGIN/../llvm/${openmp_LIB_DIR}"
+                )
+            endif()
+        endif()
+    endif()
 endif()
 
-# Add OpenMP if available
-apply_omp_settings(rocsparse-bench)
+target_link_libraries(rocsparse-bench PRIVATE
+    roc::rocsparse
+    hip::host
+    hip::device
+)
 
-# Target link libraries
-target_link_libraries(rocsparse-bench PRIVATE roc::rocsparse hip::host hip::device ${ROCSPARSE_CLIENT_LIBS})
-if (rocsparseio_FOUND)
-  target_link_libraries(rocsparse-bench PRIVATE roc::rocsparseio)
+if(OPENMP_FOUND)
+    if(TARGET OpenMP::omp)
+        target_link_libraries(rocsparse-bench PRIVATE OpenMP::omp)
+    elseif(TARGET OpenMP::OpenMP_CXX)
+        target_link_libraries(rocsparse-bench PRIVATE OpenMP::OpenMP_CXX)
+    endif()
 endif()
 
-if (NOT WIN32)
-  if (BUILD_WITH_ROCTX)
+if(rocsparseio_FOUND)
+    target_link_libraries(rocsparse-bench PRIVATE roc::rocsparseio)
+endif()
```

```suggestion
if(ROCSPARSE_ENABLE_ROCSPARSEIO)
    target_link_libraries(rocsparse-bench PRIVATE roc::rocsparseio)
endif()
```

I would like to avoid as much conditional dependencies as we can. If this doesn't break pipelines lets create an option and force users to turn it ON/OFF. This gets us closer to deterministic builds.